### PR TITLE
Optimize selects & branches & more nzcv goodies

### PR DESCRIPTION
**Pull request by alyssarosenzweig**

### CODE REVIEW

File: unittests/InstructionCountCI/FlagM/VEX_map2.json [hunk 1]
```diff
@@ -494,18 +494,23 @@
         "mrs x23, nzcv",
         "cmp x21, #0x1f (31)",
         "csel w4, w20, w22, hi",
-        "and w20, w23, #0xefffffff",
```
Comment (Sonicadvance1) [inline unittests/InstructionCountCI/FlagM/VEX_map2.json:497]: Interesting little wiggles here. Something to remember to come back to later.

**Review by Sonicadvance1 — COMMENTED** [review]
Very nice work, just a few comments.

File: unittests/InstructionCountCI/FlagM/VEX_map2.json (hunk 1 above)
Comment (alyssarosenzweig) [reply unittests/InstructionCountCI/FlagM/VEX_map2.json:497]: It's now later, deleted half of bzhi.

**Review by Sonicadvance1 — APPROVED** [review]
Nice nice nice nice.

**Pull request merged**